src/Analyzer/Passes/RewriteSumFunctionWithSumAndCountPass.cpp

```diff
@@ -0,0 +1,135 @@
+#include <Analyzer/Passes/RewriteSumFunctionWithSumAndCountPass.h>
+
+#include <AggregateFunctions/AggregateFunctionFactory.h>
+#include <Analyzer/InDepthQueryTreeVisitor.h>
+#include <Analyzer/ColumnNode.h>
+#include <Analyzer/ConstantNode.h>
+#include <Analyzer/FunctionNode.h>
+#include <Functions/FunctionFactory.h>
+
+namespace DB
+{
+
+/** Rewrite the following Analyzer to break the function `sum(column +/- literal)` into two individual functions
```

Comment:
```suggestion
/** Rewrites `sum(column +/- literal)` into two individual functions
```

`Analyzer` is the infrastructure, not the QueryTree.

Reply:
Got it. Thanks for the clarification.